api/actions/createWorkItem.js

```diff
@@ -66,87 +55,52 @@ module.exports = {
+            const buildName = milestone ? milestone.title : config.defaultBuild;
+            const foundInBuild = await Gus.resolveBuild(buildName);
+            console.log(`Found foundInBuild: ${foundInBuild} for issue titled: ${title}`);
 
-                const issue = await Gus.getByRelatedUrl(html_url);
-                const alreadyLowestPriority =
-                    issue && issue.Priority__c !== '' && issue.Priority__c <= priority;
-                const recordIdTypeIsSame = issue && issue.RecordTypeId === recordTypeId;
+            const issue = await Gus.getByRelatedUrl(html_url);
+            const bugRecordTypeId = await Gus.getBugRecordTypeId();
 
-                // if issue already exists and already has lowest priority
-                // and already has correct recordTypeId then we just return
-                if (alreadyLowestPriority && recordIdTypeIsSame) {
-                    return;
-                }
+            const alreadyLowestPriority =
+                issue && issue.Priority__c !== '' && issue.Priority__c <= priority;
+            const recordIdTypeIsSame = issue && issue.RecordTypeId === recordTypeId;
+            const isRecordTypeBug = recordTypeId === bugRecordTypeId;
 
-                if (foundInBuild) {
-                    console.log(`Found foundInBuild: ${foundInBuild} for issue titled: ${title}`);
-                    try{
-                        const syncedItem = await Gus.createWorkItemInGus(normalizedTitle,
-                            bodyInGusFormat,
-                            productTag,
-                            'NEW',
-                            foundInBuild,
-                            priority,
-                            html_url,
-                            recordTypeId);
-                        const syncedItemFromGus = await Gus.getById(syncedItem.id);
-                        const displayUrl = (hideWorkItemUrl === 'true') ? syncedItemFromGus.Name : `[${syncedItemFromGus.Name}](${process.env.WORK_ITEM_BASE_URL + syncedItem.id}/view)`;
-                        const msg = `This issue has been linked to a new work item: ${displayUrl}`;
-                        console.log(msg, ' for issue titled: ', title);
-                        return await updateIssue(req, msg);
-                    } catch(e) {
-                        console.log(`Error while creating work item ${e.message}`);
-                        return await updateIssue(req, 'Error while creating work item!');
-                    }
-                } else {
-                    console.log(`No correct build for issue titled: ${title}`);
-                    return await updateIssue(req, 'Error while creating work item. No valid build foundin GUS!');
+            // If issue is a bug we check if it already exists and already has lowest priority
+            // If issue type is investigation or story, we simply check it exists
+            if (isRecordTypeBug && alreadyLowestPriority && recordIdTypeIsSame) {
```

Comment:
This is the main fix. We don't look for priority anymore for investigation and story.